test/Transactions/Orleans.Transactions.Azure.Test/AzureTransactionLogTests.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+using Microsoft.WindowsAzure.Storage.Table;
+using Orleans.Configuration;
+using Orleans.Runtime.Configuration;
+using Orleans.Transactions.Abstractions;
+using TestExtensions;
+using Xunit;
+
+namespace Orleans.Transactions.AzureStorage.Tests
+{
+    [TestCategory("Azure"), TestCategory("Transactions"), TestCategory("BVT")]
+    public class AzureTransactionLogTests 
+    {
+        private const string TestClusterId = "AzureTransactionLogTestCluster";
+        private static string TableName = $"TransactionLog{((uint) Guid.NewGuid().GetHashCode()) % 100000}";
+        public AzureTransactionLogTests()
+        {
+            TestFixture.CheckForAzureStorage(TestDefaultConfiguration.DataConnectionString);
+        }
+
+        [SkippableFact]
+        public async Task TransactionLogCanArchiveAndQuery()
+        {
+            var azureOptions = new AzureTransactionLogOptions()
+            {
+                ArchiveLog = true,
+                ConnectionString = TestDefaultConfiguration.DataConnectionString,
+                TableName = TableName
+            };
+
+            var logStorage = await StorageFactory(azureOptions);
+            var recordsNum = 2000;
+            var allTransactions = new List<CommitRecord>(recordsNum);
+            for (int i = 0; i< recordsNum; i++)
+            {
+                allTransactions.Add(new CommitRecord(){LSN=i, TransactionId = i});
+            }
+            await logStorage.Append(allTransactions);
+            //a subset of all transactions will be archived 
+            await logStorage.TruncateLog(recordsNum + 2);
```

Comment:
unclear why this would archive only a subset?

Reply:
because it might now archive all . For example, if the batch operation didn't hit 100 operation count, the batch operation won't execute

Reply:
That sounds like a bug...

Reply:
yeah... you are right... it was a bug in the old code.. I wasn't thinking that is a bug because I assume the original code is bugless..

so log would only be truncated when the   if (batchOperation.Count == BatchOperationLimit)